target/mips/translate.c

```diff
@@ -6554,30 +6574,9 @@ static void gen_compute_branch(DisasContext *ctx, uint32_t opc,
 #ifndef TARGET_CHERI
     case OPC_JALX:
 #endif
-        /* Jump to immediate */
-
-#ifdef TARGET_CHERI
-        /*
-         * The branch target address is not known since the PCC.base can
-         * change.  Therefore treat this like a JR/JALR and compute the
-         * branch target from the current PCC.base.
-         *
-         * ((ctx->base.pc_next + insn_bytes) - PCC.base) >> 28 << 28 | offset +
-         * PCC.base
-         */
-        tcg_gen_movi_i64(t0, ctx->base.pc_next + insn_bytes);
-        tcg_gen_ld_i64(t1, cpu_env, offsetof(CPUMIPSState, active_tc.PCC) +
-            offsetof(cap_register_t, cr_base));
-        tcg_gen_sub_i64(t0, t0, t1);
-        tcg_gen_shri_i64(t0, t0, 28);
-        tcg_gen_shli_i64(t0, t0, 28);
-        tcg_gen_ori_i64(t0, t0, offset);
-        tcg_gen_add_i64(btarget, t0, t1);
-        GEN_CCHECK_BTARGET(cpu_env);
-#else
-        btgt = ((ctx->base.pc_next + insn_bytes) & (int32_t)0xF0000000) |
-            (uint32_t)offset;
-#endif /* ! TARGET_CHERI */
+        /* Jump to immediate (taking PCC.base into account) */
+        btgt = (((ctx->base.pc_next - pcc_base(ctx) + insn_bytes) &
```

Comment:
Put the base subtraction last to make it consistent with the BLTZALL and blink cases?